src/snowcli/cli/nativeapp/manager.py

```diff
@@ -219,12 +219,18 @@ def get_app_pkg_distribution_in_snowflake(self) -> str:
             )
         )
 
-    def is_app_pkg_distribution_same_in_sf(self) -> bool:
+    def check_app_pkg_distribution_against_resolved_definition(
```

Comment:
how about simply: `verify_project_distribution(expected_distribution: Optional[str)`

Without an argument, it checks the project distribution against SF. With an arg, it verifies against that instead.

Not sure what a "resolved definition" means at first glance. I like verify here because it implies that matching is very much expected, so issuing a warning in this method makes sense.

Reply:
I like that way of reasoning, updating now!